pkg/util/annotation.go

```diff
@@ -18,7 +26,15 @@ func MergeAnnotation(obj *unstructured.Unstructured, annotationKey string, annot
 // MergeAnnotations merges the annotations from 'src' to 'dst', identical keys will not be merged.
 func MergeAnnotations(dst *unstructured.Unstructured, src *unstructured.Unstructured) {
 	for key, value := range src.GetAnnotations() {
-		MergeAnnotation(dst, key, value)
+		objectAnnotation := dst.GetAnnotations()
+		if objectAnnotation == nil {
+			objectAnnotation = make(map[string]string, 1)
+		}
+		deletedAnnotationKeys := getDeletedAnnotationKeys(dst, src)
+		if _, exist := objectAnnotation[key]; !exist && !deletedAnnotationKeys.Has(key) {
+			objectAnnotation[key] = value
+			dst.SetAnnotations(objectAnnotation)
+		}
```

Comment:
How about update like this:
```diff
 func MergeAnnotations(dst *unstructured.Unstructured, src *unstructured.Unstructured) {
+       objectAnnotation := dst.GetAnnotations()
+       if objectAnnotation == nil {
+               objectAnnotation = make(map[string]string, 1)
+       }
+
+       deletedAnnotationKeys := getDeletedAnnotationKeys(dst, src)
+
        for key, value := range src.GetAnnotations() {
-               objectAnnotation := dst.GetAnnotations()
-               if objectAnnotation == nil {
-                       objectAnnotation = make(map[string]string, 1)
-               }
-               deletedAnnotationKeys := getDeletedAnnotationKeys(dst, src)
                if _, exist := objectAnnotation[key]; !exist && !deletedAnnotationKeys.Has(key) {
                        objectAnnotation[key] = value
-                       dst.SetAnnotations(objectAnnotation)
                }
        }
+
+       dst.SetAnnotations(objectAnnotation)
 }
```

This can reduce the amount of computation.